File: python examples/Arcade Drive/subsystems/chassis.py
from wpilib.command import Subsystem
from ctre import WPI_TalonSRX
from commands.arcadeDrive import ArcadeDrive

class Chassis (wpilib.command.Subsystem):

  def __init__ (self, robot):
    self.robot = robot

    self.leftMaster = WPI_TalonSRX(50)
    self.leftSlaveA = WPI_TalonSRX(51)
    self.leftSlaveB = WPI_TalonSRX(52)

    self.rightMaster = WPI_TalonSRX(40)
    self.rightSlaveA = WPI_TalonSRX(41)
    self.rightSlaveB = WPI_TalonSRX(42)

    self.leftMaster.setInverted(True)
    self.leftSlaveA.setInverted(True)
    self.leftSlaveB.setInverted(True)

    self.rightMaster.setInverted(False)
    self.rightSlaveA.setInverted(False)
    self.rightSlaveB.setInverted(False)

    self.leftSlaveA.follow(self.leftMaster)
    self.leftSlaveB.follow(self.leftMaster)

    self.rightSlaveA.follow(self.rightMaster)
    self.rightSlaveB.follow(self.rightMaster)

  def setLeftRaw (rate):
    self.leftMaster.set(rate)

  def setRightRaw (rate):
    self.rightMaster.set(rate)

  def fullStop ():
    self.leftMaster.set(0.0)
    self.rightMaster.set(0.0)

  def initDefaultCommand ():
    self.setDefaultCommand(ArcadeDrive(self.robot))

File: c++ examples/Arcade Drive/cpp/impl/subsystems/Chassis.cc
#include "WPILib.h"
#include "../../include/subsystems/Chassis.hh"
#include "../commands/ArcadeDrive.cc"

Chassis::Chassis (Robot* robot) {
  this->leftMaster = new frc::Spark(1);
  this->leftSlaveA = new frc::Spark(2);
  this->leftSlaveB = new frc::Spark(3);

  this->rightMaster = new frc::Spark(4);
  this->rightSlaveA = new frc::Spark(5);
  this->rightSlaveB = new frc::Spark(6);

  this->robot = robot;
}

void Chassis::SetLeftRaw (double rate) const {
  this->leftMaster->Set(rate);
  this->leftSlaveA->Set(rate);
  this->leftSlaveB->Set(rate);
}

void Chassis::SetRightRaw (double rate) const {
  this->rightMaster->Set(rate);
  this->rightSlaveA->Set(rate);
  this->rightSlaveB->Set(rate);
}

void Chassis::InitDefaultCommand () {
  SetDefaultCommand(new ArcadeDrive(this->robot));
}